src/bci_build/package/gcc.py

```diff
@@ -42,6 +50,14 @@ def _is_main_gcc(os_version: OsVersion, gcc_version: _GCC_VERSIONS) -> bool:
         package_name=f"gcc-{gcc_version}-image",
         os_version=os_version,
         version=gcc_version,
+        support_level=(
+            SupportLevel.L3 if not os_version.is_sle15 else SupportLevel.TECHPREVIEW
+        ),
+        supported_until=(
+            _GCC_SLCC_SUPPORTED_UNTIL.get(gcc_version)
+            if not os_version.is_sle15
```

Comment:
Is the not intentional?
```suggestion
            if not os_version.is_sle15
```

Reply:
Yes, the lifecycle of gcc is different for sle15 (its 6 months after x+1.2 got released on sle15). As that is impossible to calculate upfront, we need to leave it as None for sle15.

The lifecycle above is the gcc slcc lifecycle which can be calculated upfront because it's keyed on the upstream release date

Reply:
That's why the dict has _SLCC_ in the name